src/CoreWCF.WebHttp/src/CoreWCF/Dispatcher/ContentTypeSettingDispatchMessageFormatter.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using CoreWCF.Channels;
+using CoreWCF.Web;
+
+namespace CoreWCF.Dispatcher
+{
+    internal class ContentTypeSettingDispatchMessageFormatter : IDispatchMessageFormatter
+    {
+        private readonly IDispatchMessageFormatter _innerFormatter;
+        private readonly string _outgoingContentType;
+
+        public ContentTypeSettingDispatchMessageFormatter(string outgoingContentType, IDispatchMessageFormatter innerFormatter)
+        {
+            _outgoingContentType = outgoingContentType ?? throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(outgoingContentType));
+            _innerFormatter = innerFormatter ?? throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(innerFormatter));
+        }
+
+        public void DeserializeRequest(Message message, object[] parameters)
+        {
+            _innerFormatter.DeserializeRequest(message, parameters);
+        }
+
+        public Message SerializeReply(MessageVersion messageVersion, object[] parameters, object result)
+        {
+            Message message = _innerFormatter.SerializeReply(messageVersion, parameters, result);
+            if (message != null)
+            {
+                AddResponseContentTypeProperty(message, _outgoingContentType);
+            }
+
+            return message;
+        }
+
+        static void AddResponseContentTypeProperty(Message message, string contentType)
+        {
```

Comment:
Accessibility modifier

Reply:
This was addressed in 4671ecfb13a7628fd4bcf310abd35993645539ca.